From 449f0a85795a07d1e1ae07e4d9a434628a0b5ac4 Mon Sep 17 00:00:00 2001
From: Kostadin Shishmanov <kostadinshishmanov@protonmail.com>
Date: Fri, 4 Sep 2026 13:23:17 +0300
Subject: [PATCH] aco: add missing <algorithm> include

aco_util.h uses std::move_backward but doesn't include <algorithm>
which causes a build failure with libc++ 23 since it's dropping such
transitive includes.

Cc: mesa-stable
Signed-off-by: Kostadin Shishmanov <kostadinshishmanov@protonmail.com>
Reviewed-by: Georg Lehmann <dadschoorse@gmail.com>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/44233>
---
 src/amd/compiler/aco_util.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/amd/compiler/aco_util.h b/src/amd/compiler/aco_util.h
index 82d754f155aa..9f20e80a5545 100644
--- a/src/amd/compiler/aco_util.h
+++ b/src/amd/compiler/aco_util.h
@@ -12,6 +12,7 @@
 #include "util/macros.h"
 #include "util/u_math.h"
 
+#include <algorithm>
 #include <array>
 #include <cassert>
 #include <cstddef>
-- 
GitLab

